components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java

```diff
@@ -429,6 +429,13 @@ public boolean handleRequest(MessageContext messageContext) {
             } else if (Util.tracingEnabled()) {
                 Util.setTag(keyTracingSpan, APIMgtGatewayConstants.ERROR, APIMgtGatewayConstants.KEY_SPAN_ERROR);
             }
+
+            // Get application information to be logged in API authentication failure (only requestURI for JWT tokens)
+            String applicationName = (String) messageContext.getProperty(APIMgtGatewayConstants.APPLICATION_NAME);
+            String endUserName = (String) messageContext.getProperty(APIMgtGatewayConstants.END_USER_NAME);
+            String requestURI = (String) messageContext.getProperty(RESTConstants.REST_FULL_REQUEST_PATH);
+            String warnDetails = "";
+
             if (log.isDebugEnabled()) {
                     // We do the calculations only if the debug logs are enabled. Otherwise this would be an overhead
```

Comment:
shall we move the construction of warn log message to a separate method?

Reply:
added a separate method and included the sample error strings